### PR TITLE
feat: ignore node_modules folders when resolving components glob

### PR DESCRIPTION
**Pull request by dstaley**
This PR ignores `node_modules` folders when resolving the components glob, which allows for nested components in a single package.

### CODE REVIEW

File: components-loader.js [hunk 1]
```diff
@@ -10,9 +10,13 @@ module.exports = function swingsetComponentsLoader() {
   const { pluginOptions, webpackConfig } = getOptions(this)
 
   // Resolve components glob
-  const allComponents = globby.sync(`${pluginOptions.componentsRoot}`, {
-    onlyFiles: false,
-  })
+  const allComponents = globby
+    .sync(`${pluginOptions.componentsRoot}`, {
+      onlyFiles: false,
+    })
```
Comment (BRKalow) [inline components-loader.js:16]: 💭 [idea] could we use the `gitignore` option and/or the `ignoreFiles` option here? https://github.com/sindresorhus/globby#gitignore

Comment (dstaley) [reply components-loader.js:16]: My only concern would be for users who are applying a compile-step to their components (for example, from TS -> JS). Using `.gitignore` might be too wide a net when we really just want to make sure we're not treating a nested `node_modules` folder as a component.

That being said, maybe we shift that burden to the user by removing this and just making sure that `componentsRoot` supports arrays with negated patterns: `['packages/**/*', '!packages/**/node_modules']`

Comment (BRKalow) [reply components-loader.js:16]: Fair point on `.gitignore`. Excluding `node_modules` by default feels reasonable though!

**Comment by dstaley** [comment]
@BRKalow do you think this is good to merge + release, or should we keep running a canary release with this for a while longer?

**Pull request merged**